diff -ruN bcache-tools-20130827.orig/bcache-register bcache-tools-20130827/bcache-register
--- bcache-tools-20130827.orig/bcache-register	2013-08-26 23:46:19.000000000 +0200
+++ bcache-tools-20130827/bcache-register	2013-08-27 08:49:59.924940825 +0200
@@ -1,4 +1,9 @@
 #!/bin/sh
-modprobe -qba bcache
-test -f /sys/fs/bcache/register_quiet && echo "$1" > /sys/fs/bcache/register_quiet
+#
+# Utility script to be called from udev, so it's not monkey proof at all. It's
+# sole purpose is to load the bcache kernel module whenever a bcache device is
+# detected.
+#
+/sbin/modprobe -qba bcache
+test -f /sys/fs/bcache/register_quiet && echo "$1" > /sys/fs/bcache/register_quiet 2>/dev/null
 
