--- bcache-tools-20130909/61-bcache.rules.udevfix	2013-09-11 21:39:09.690473058 +0200
+++ bcache-tools-20130909/61-bcache.rules	2013-09-11 21:38:49.273376220 +0200
@@ -5,13 +5,12 @@
 ACTION=="remove", GOTO="bcache_end"
 
 # Backing devices: scan, symlink, register
-IMPORT{program}="/sbin/blkid -o udev $tempnode"
 # blkid and probe-bcache can disagree, in which case don't register
 ENV{ID_FS_TYPE}=="?*", ENV{ID_FS_TYPE}!="bcache", GOTO="bcache_backing_end"
 
 IMPORT{program}="/sbin/probe-bcache -o udev $tempnode"
-ENV{ID_FS_UUID_ENC}=="?*", SYMLINK+="disk/by-uuid/$env{ID_FS_UUID_ENC}"
-SUBSYSTEM=="block", ACTION=="add|change", ENV{ID_FS_TYPE}=="bcache", \
+ENV{BCACHE_ID_FS_UUID_ENC}=="?*", SYMLINK+="disk/by-uuid/$env{BCACHE_ID_FS_UUID_ENC}"
+SUBSYSTEM=="block", ACTION=="add|change", ENV{BCACHE_ID_FS_TYPE}=="bcache", \
         RUN+="bcache-register $tempnode"
 LABEL="bcache_backing_end"
 
