--- a/src/libvirt.c	2009-08-11 23:27:35.000000000 +0200
+++ b/src/libvirt.c	2009-08-12 00:14:58.000000000 +0200
@@ -49,6 +49,8 @@
 
 /* Connection. */
 static virConnectPtr conn = 0;
+static char *conn_string = NULL;
+static int conn_count = 0;
 
 /* Seconds between list refreshes, 0 disables completely. */
 static int interval = 60;
@@ -153,14 +155,14 @@
         il_interface_devices = ignorelist_create (1);
 
     if (strcasecmp (key, "Connection") == 0) {
-        if (conn != 0) {
+        if (conn_count++ != 0) {
             ERROR ("Connection may only be given once in config file");
             return 1;
         }
-        conn = virConnectOpenReadOnly (value);
-        if (!conn) {
-            VIRT_ERROR (NULL, "connection failed");
-            return 1;
+        conn_string = strdup(value);
+        if (conn_string == NULL) {
+            ERROR ("libvirt plugin: Connection strdup failed.");
+            return -1;
         }
         return 0;
     }
@@ -253,10 +255,11 @@
     int i;
 
     if (conn == NULL) {
-        ERROR ("libvirt plugin: Not connected. Use Connection in "
-                "config file to supply connection URI.  For more information "
-                "see <http://libvirt.org/uri.html>");
-        return -1;
+        conn = virConnectOpenReadOnly (conn_string);
+	if (conn == NULL) {
+	    ERROR ("libvirt plugin: Not connected.");
+	    return -1;
+	}
     }
 
     time (&t);
@@ -264,8 +267,12 @@
     /* Need to refresh domain or device lists? */
     if ((last_refresh == (time_t) 0) ||
             ((interval > 0) && ((last_refresh + interval) <= t))) {
-        if (refresh_lists () != 0)
+        if (refresh_lists () != 0) {
+            if (conn != NULL)
+                virConnectClose (conn);
+            conn = NULL;
             return -1;
+        }
         last_refresh = t;
     }
 
