--- ImageMagick-6.1.8/Magick++/lib/ImageMagick++.pc.in.purge	2003-11-01 14:02:49.000000000 -0600
+++ ImageMagick-6.1.8/Magick++/lib/ImageMagick++.pc.in	2005-01-18 12:55:01.000000000 -0600
@@ -8,5 +8,5 @@
 Version: @PACKAGE_VERSION@
 Description: Magick++ - Convert, Edit, and Compose Images
 Requires: ImageMagick
-Libs: -L${libdir} @MAGICK_LDFLAGS@ @MAGICK_LIBS@ -lMagick++
-Cflags: -I${includedir} @MAGICK_CPPFLAGS@
+Libs: -L${libdir} -lMagick++
+Cflags: -I${includedir} 
