--- ImageMagick-6.2.0/config/delegates.xml.in.lprhack	2005-03-16 09:27:05.564324000 -0500
+++ ImageMagick-6.2.0/config/delegates.xml.in	2005-03-16 09:28:31.015788000 -0500
@@ -89,7 +89,7 @@
     "@ConvertDelegate@" -concatenate "%o*.png" "%o"' />
   <delegate decode="ps" encode="eps" mode="bi" command='"@PSDelegate@" -q -dBATCH -dSAFER -dMaxBitmap=500000000 -dNOPAUSE -dAlignToPixels=0 -sDEVICE="@GSEPSDevice@" -sOutputFile="%o" -f"%i"' />
   <delegate decode="ps" encode="pdf" mode="bi" command='"@PSDelegate@" -q -dBATCH -dSAFER -dMaxBitmap=500000000 -dNOPAUSE -dAlignToPixels=0 -sDEVICE="@GSPDFDevice@" -sOutputFile="%o" -f"%i"' />
-  <delegate decode="ps" encode="print" mode="encode" command='"@PrintDelegate@" "%i"' />
+  <delegate decode="ps" encode="print" mode="encode" command='lpr "%i"' />
   <delegate decode="pxl" command='"@PCLDelegate@" -dBATCH -dSAFER -dNOPAUSE -dAlignToPixels=0 -sDEVICE=pdfwrite -sOutputFile="%o" "%i"' />
   <delegate decode="rad" command='"@RADDecodeDelegate@" -g 1.0 "%i" "%o"' />
   <delegate decode="rgba" encode="rle" mode="encode" command='"@MogrifyDelegate@" -flip -size %wx%h "rgba:%i"
