--- ImageMagick-6.2.1/magick/xwindow.c.fixed	2005-04-25 01:10:47.727810000 -0400
+++ ImageMagick-6.2.1/magick/xwindow.c	2005-04-25 01:11:01.698825000 -0400
@@ -779,8 +779,8 @@
     },
     *TextFonts[]=
     {
-      "-*-fixed-medium-r-normal-*-12-*-*-*-*-*-iso8859-1",
-      "-*-fixed-medium-r-normal-*-12-*-*-*-*-*-iso8859-15",
+      "-*-courier-medium-r-normal-*-12-*-*-*-*-*-iso8859-1",
+      "-*-courier-medium-r-normal-*-12-*-*-*-*-*-iso8859-15",
       "fixed",
       (char *) NULL
     };
