diff -up ImageMagick-6.4.9-6/magick/magick-type.h~ ImageMagick-6.4.9-6/magick/magick-type.h
--- ImageMagick-6.4.9-6/magick/magick-type.h~	2009-01-15 03:13:58.000000000 +0100
+++ ImageMagick-6.4.9-6/magick/magick-type.h	2009-03-13 10:30:59.000000000 +0100
@@ -127,7 +127,7 @@ typedef unsigned __int64 MagickSizeType;
 #define MagickSizeFormat  "%10llu"
 #endif
 
-#if _MSC_VER == 1200
+#if defined(_MSC_VER) && (_MSC_VER == 1200)
 typedef MagickOffsetType QuantumAny;
 #else
 typedef MagickSizeType QuantumAny;
