diff -up ImageMagick-6.4.0/configure.ac.orig ImageMagick-6.4.0/configure.ac
--- ImageMagick-6.4.0/configure.ac.orig	2008-04-22 05:38:56.000000000 +0200
+++ ImageMagick-6.4.0/configure.ac	2008-04-27 16:33:28.000000000 +0200
@@ -2391,7 +2391,7 @@ AC_MSG_RESULT([-------------------------
 AutotraceDecodeDelegateDefault='autotrace'
 AVIDecodeDelegateDefault='mplayer'
 BZIPDelegateDefault='bzip2'
-BrowseDelegateDefault='htmlview'
+BrowseDelegateDefault='xdg-open'
 CGMDecodeDelegateDefault='ralcgm'
 CatDelegateDefault='cat'
 DNGDecodeDelegateDefault='ufraw-batch'
diff -up ImageMagick-6.4.0/magick/delegate.c.orig ImageMagick-6.4.0/magick/delegate.c
--- ImageMagick-6.4.0/magick/delegate.c.orig	2008-04-06 03:39:42.000000000 +0200
+++ ImageMagick-6.4.0/magick/delegate.c	2008-04-27 16:35:46.000000000 +0200
@@ -78,7 +78,7 @@ static const char
     "<delegatemap>"
     "  <delegate decode=\"autotrace\" stealth=\"True\" command=\"&quot;autotrace&quot; -output-format svg -output-file &quot;%o&quot; &quot;%i&quot;\"/>"
     "  <delegate decode=\"avi:decode\" stealth=\"True\" command=\"&quot;mplayer&quot; &quot;%i&quot; -really-quiet -ao null -vo png:z=3\"/>"
-    "  <delegate decode=\"browse\" stealth=\"True\" spawn=\"True\" command=\"&quot;htmlview&quot; http://www.imagemagick.org/\"/>"
+    "  <delegate decode=\"browse\" stealth=\"True\" spawn=\"True\" command=\"&quot;xdg-open&quot; http://www.imagemagick.org/\"/>"
     "  <delegate decode=\"cgm\" thread-support=\"False\" command=\"&quot;ralcgm&quot; -d ps -oC &lt; &quot;%i&quot; &gt; &quot;%o&quot; &gt;&gt;/dev/null\"/>"
     "  <delegate decode=\"dng:decode\" command=\"&quot;/usr/bin/ufraw-batch&quot; --silent --wb=camera --black-point=auto --exposure=auto --create-id=also --out-type=ppm16 &quot;--output=%u.pnm&quot; &quot;%i&quot;\"/>"
     "  <delegate decode=\"edit\" stealth=\"True\" command=\"&quot;xterm&quot; -title &quot;Edit Image Comment&quot; -e vi &quot;%o&quot;\"/>"
diff -up ImageMagick-6.4.0/www/source/delegates.xml.orig ImageMagick-6.4.0/www/source/delegates.xml
--- ImageMagick-6.4.0/www/source/delegates.xml.orig	2008-04-04 18:43:27.000000000 +0200
+++ ImageMagick-6.4.0/www/source/delegates.xml	2008-04-27 16:35:06.000000000 +0200
@@ -57,7 +57,7 @@
 -->
 <delegatemap>
   <delegate decode="autotrace" stealth="True" command="&quot;autotrace&quot; -output-format svg -output-file &quot;%o&quot; &quot;%i&quot;"/>
-  <delegate decode="browse" stealth="True" spawn="True" command="&quot;htmlview&quot; http://www.imagemagick.org/"/>
+  <delegate decode="browse" stealth="True" spawn="True" command="&quot;xdg-open&quot; http://www.imagemagick.org/"/>
   <delegate decode="cgm" thread-support="False" command="&quot;ralcgm&quot; -d ps -oC &lt; &quot;%i&quot; &gt; &quot;%o&quot; 2&gt; &quot;%u&quot;"/>
   <delegate decode="dng:decode" command="&quot;ufraw-batch&quot; --silent --wb=camera --black-point=auto --exposure=auto --create-id=also --out-type=ppm16 &quot;--output=%u.pnm&quot; &quot;%i&quot;"/>
   <delegate decode="edit" stealth="True" command="&quot;xterm&quot; -title &quot;Edit Image Comment&quot; -e vi &quot;%o&quot;"/>
diff -up ImageMagick-6.4.0/configure.orig ImageMagick-6.4.0/configure
--- ImageMagick-6.4.0/configure.orig	2008-04-24 17:07:05.000000000 +0200
+++ ImageMagick-6.4.0/configure	2008-04-27 16:33:28.000000000 +0200
@@ -40762,7 +40762,7 @@ $as_echo "------------------------------
 AutotraceDecodeDelegateDefault='autotrace'
 AVIDecodeDelegateDefault='mplayer'
 BZIPDelegateDefault='bzip2'
-BrowseDelegateDefault='htmlview'
+BrowseDelegateDefault='xdg-open'
 CGMDecodeDelegateDefault='ralcgm'
 CatDelegateDefault='cat'
 DNGDecodeDelegateDefault='ufraw-batch'
