--- ImageMagick-6.3.2/PerlMagick/Makefile.PL.orig	2007-02-25 03:01:28.000000000 +1000
+++ ImageMagick-6.3.2/PerlMagick/Makefile.PL	2007-03-07 14:19:21.000000000 +1000
@@ -62,7 +62,7 @@ WriteMakefile
    'INSTALLBIN'	=> '/usr/bin',
 
    # Library specification
-   'LIBS'	=> [ '-L/usr/lib -lfreetype -llcms -ltiff -lfreetype -ljpeg -lfontconfig -lXext -lSM -lICE -lX11 -lXt -lbz2 -lz -lpthread -lm -lpthread' ],
+   'LIBS'	=> [ '-L/usr/lib -L../magick/.libs -lMagick -lfreetype -llcms -ltiff -lfreetype -ljpeg -lfontconfig -lXext -lSM -lICE -lX11 -lXt -lbz2 -lz -lpthread -lm -lpthread' ],
 
    # Perl binary name (if a Perl binary is built)
    'MAP_TARGET'	=> 'PerlMagick',
--- ImageMagick-6.3.2/PerlMagick/Makefile.PL.in.orig	2006-09-20 00:53:54.000000000 +1000
+++ ImageMagick-6.3.2/PerlMagick/Makefile.PL.in	2007-03-07 14:20:11.000000000 +1000
@@ -62,7 +62,7 @@ WriteMakefile
    'INSTALLBIN'	=> '@BIN_DIR@',
 
    # Library specification
-   'LIBS'	=> [ '-L@MAGICKLIBDIR@ @LDFLAGS@ @MAGICK_DEP_LIBS@' ],
+   'LIBS'	=> [ '-L@MAGICKLIBDIR@ -L../magick/.libs -lMagick @LDFLAGS@ @MAGICK_DEP_LIBS@' ],
 
    # Perl binary name (if a Perl binary is built)
    'MAP_TARGET'	=> 'PerlMagick',
