--- ImageMagick-6.2.0/magick/studio.h.vsnprintf	2005-03-18 12:36:54.000000000 -0500
+++ ImageMagick-6.2.0/magick/studio.h	2005-03-18 12:37:02.000000000 -0500
@@ -175,11 +175,6 @@
   strlcpy(char *,const char *,size_t);
 #endif
 
-#if defined(HAVE_VSNPRINTF) && !defined(HAVE_VSNPRINTF_PROTOTYPE)
-extern int
-  vsnprintf(char *,size_t,const char *,va_list);
-#endif
-
 #if !defined(magick_attribute)
 #  if !defined(__GNUC__)
 #    define magick_attribute(x)  /* nothing */
