--- ImageMagick-6.1.8/magick/version.h.in.local_doc	2004-12-17 18:28:02.000000000 -0600
+++ ImageMagick-6.1.8/magick/version.h.in	2005-01-18 10:42:27.712406016 -0600
@@ -28,7 +28,7 @@
 #define MagickReleaseDate  "@PACKAGE_RELEASE_DATE@"
 #define MagickVersion MagickPackageName " " MagickLibVersionText " " \
   MagickReleaseDate " " MagickQuantumDepth " " MagickHomeURL
-#define MagickHomeURL  "http://www.imagemagick.org"
+#define MagickHomeURL  "file:/usr/share/ImageMagick-@PACKAGE_VERSION@/doc/index.html"
 
 extern MagickExport char
   *GetMagickHomeURL(void);
