--- ImageMagick-6.2.8/magick/Magick-config.in.multilib	2006-05-26 21:58:55.000000000 -0400
+++ ImageMagick-6.2.8/magick/Magick-config.in	2006-06-09 10:12:09.000000000 -0400
@@ -15,28 +15,28 @@
 while test $# -gt 0; do
   case $1 in
     --prefix)
-      echo @PREFIX_DIR@
+      pkg-config --variable prefix ImageMagick
       ;;
     --exec-prefix)
-      echo @EXEC_PREFIX_DIR@
+      pkg-config --variable exec_prefix ImageMagick
       ;;
     --version)
-      echo @PACKAGE_VERSION@
+      pkg-config --modversion ImageMagick
       ;;
     --cflags)
-      echo '@MAGICK_CFLAGS@'
+      pkg-config --cflags ImageMagick
       ;;
     --cxxflags)
-      echo '@MAGICK_CXXFLAGS@'
+      pkg-config --cflags ImageMagick
       ;;
     --cppflags)
-      echo '@MAGICK_CPPFLAGS@'
+      pkg-config --cflags ImageMagick
       ;;
     --ldflags)
-      echo '@MAGICK_LDFLAGS@'
+      pkg-config --libs ImageMagick
       ;;
     --libs)
-      echo '@MAGICK_LIBS@ -lWand -lMagick'
+      pkg-config --libs ImageMagick
       ;;
     *)
       echo "${usage}" 1>&2
--- ImageMagick-6.2.8/wand/Wand-config.in.multilib	2006-05-26 21:58:55.000000000 -0400
+++ ImageMagick-6.2.8/wand/Wand-config.in	2006-06-09 10:16:55.000000000 -0400
@@ -15,28 +15,28 @@
 while test $# -gt 0; do
   case $1 in
     --prefix)
-      echo @PREFIX_DIR@
+      pkg-config --variable prefix Wand
       ;;
     --exec-prefix)
-      echo @EXEC_PREFIX_DIR@
+      pkg-config --variable exec_prefix Wand
       ;;
     --version)
-      echo @PACKAGE_VERSION@
+      pkg-config --modversion Wand
       ;;
     --cflags)
-      echo '@MAGICK_CFLAGS@'
+      pkg-config --cflags Wand
       ;;
     --cxxflags)
-      echo '@MAGICK_CXXFLAGS@'
+      pkg-config --cflags Wand
       ;;
     --cppflags)
-      echo '@MAGICK_CPPFLAGS@'
+      pkg-config --cflags Wand
       ;;
     --ldflags)
-      echo '@MAGICK_LDFLAGS@'
+      pkg-config --libs Wand
       ;;
     --libs)
-      echo '@MAGICK_LIBS@ -lWand -lMagick'
+      pkg-config --libs Wand
       ;;
     *)
       echo "${usage}" 1>&2
