--- /repository/pecl/memcached/php_memcached.c	2009/06/04 22:40:49	1.31
+++ /repository/pecl/memcached/php_memcached.c	2009/06/22 16:07:13	1.32
@@ -2802,9 +2802,18 @@
 	 * Indicate whether igbinary serializer is available
 	 */
 #ifdef HAVE_MEMCACHED_IGBINARY
-	REGISTER_MEMC_CLASS_CONST_LONG(IGBINARY_SUPPORT, 1);
+	REGISTER_MEMC_CLASS_CONST_LONG(HAVE_IGBINARY, 1);
 #else
-	REGISTER_MEMC_CLASS_CONST_LONG(IGBINARY_SUPPORT, 0);
+	REGISTER_MEMC_CLASS_CONST_LONG(HAVE_IGBINARY, 0);
+#endif
+
+	/*
+	 * Indicate whether json serializer is available
+	 */
+#ifdef HAVE_JSON_API
+	REGISTER_MEMC_CLASS_CONST_LONG(HAVE_JSON, 1);
+#else
+	REGISTER_MEMC_CLASS_CONST_LONG(HAVE_JSON, 0);
 #endif
 
 	/*
