--- GNUmakefile.old	2007-07-02 17:13:05.000000000 +0300
+++ GNUmakefile	2007-07-02 17:13:59.000000000 +0300
@@ -4,13 +4,13 @@
 #
 #-----------------------------------------------------
 
-all: Makefile.config liblwgeom loaderdumper utils templategis
+all: Makefile.config liblwgeom loaderdumper utils
 
-install: all liblwgeom-install loaderdumper-install templategis-install
+install: all liblwgeom-install loaderdumper-install
 
-uninstall: liblwgeom-uninstall loaderdumper-uninstall docs-uninstall templategis-uninstall
+uninstall: liblwgeom-uninstall loaderdumper-uninstall docs-uninstall
 
-clean: Makefile.config liblwgeom-clean loaderdumper-clean docs-clean test-clean templategis-clean
+clean: Makefile.config liblwgeom-clean loaderdumper-clean docs-clean test-clean
 	rm -f lwpostgis.sql lwpostgis_upgrade.sql
 
 distclean: clean
@@ -59,18 +59,6 @@
 loaderdumper-uninstall:
 	$(MAKE) -C loader uninstall
 
-templategis: Makefile.config
-	$(MAKE) -C extras/template_gis
-
-templategis-clean:
-	$(MAKE) -C extras/template_gis clean
-
-templategis-install:
-	$(MAKE) -C extras/template_gis install
-
-templategis-uninstall:
-	$(MAKE) -C extras/template_gis uninstall
-
 docs: Makefile.config
 	$(MAKE) -C doc 
 
