--- java/jdbc/Makefile.old	2007-12-06 15:18:04.000000000 -0800
+++ java/jdbc/Makefile	2007-12-06 15:22:19.000000000 -0800
@@ -27,9 +27,9 @@
 
 # Configure the helper executables used during build.
 
-JAVAC?=javac -target 1.2 -source 1.2
+JAVAC?=ecj -target 1.5 -source 1.5
 JAVA?=java
-JAR?=jar
+JAR?=fastjar
 MKDIR?=mkdir -p
 DELETE?=rm -rvf
 CP?=cp
@@ -264,7 +264,7 @@
 	$(MKDIR) javadoc-build
 	# We use an ugly trick to build on windows and unix - 
 	# javadoc command line options are not portable wr/t path separators
-	javadoc -d javadoc-build -sourcepath "src:jtssrc:;src;jtssrc" \
+	sinjdoc -d javadoc-build -sourcepath "src:jtssrc:;src;jtssrc" \
 		org.postgis org.postgis.jts org.postgis.binary org.postgis.java2d examples
 
 # Preliminary JTS support
