--- bcache-tools-20131018/69-bcache.rules.orig	2013-10-18 19:33:45.991178453 +0200
+++ bcache-tools-20131018/69-bcache.rules	2013-10-18 19:34:38.907431161 +0200
@@ -3,6 +3,7 @@
 
 SUBSYSTEM!="block", GOTO="bcache_end"
 ACTION=="remove", GOTO="bcache_end"
+ENV{DM_UDEV_DISABLE_OTHER_RULES_FLAG}=="1", GOTO="bcache_end"
 
 # blkid was run by the standard udev rules
 # It recognised bcache (util-linux 2.24+)
