--- bcache-tools-20130909/61-bcache.rules.udevfix-old	2013-09-01 12:38:11.000000000 +0200
+++ bcache-tools-20130909/61-bcache.rules	2013-09-30 19:09:34.457450713 +0200
@@ -4,14 +4,16 @@
 SUBSYSTEM!="block", GOTO="bcache_end"
 ACTION=="remove", GOTO="bcache_end"
 
+# There's no certainty that blkid was executed by any of the earlier rules
+# so to be sure it's executed here too
+IMPORT{builtin}="blkid"
 # Backing devices: scan, symlink, register
-IMPORT{program}="/sbin/blkid -o udev $tempnode"
 # blkid and probe-bcache can disagree, in which case don't register
 ENV{ID_FS_TYPE}=="?*", ENV{ID_FS_TYPE}!="bcache", GOTO="bcache_backing_end"
 
 IMPORT{program}="/sbin/probe-bcache -o udev $tempnode"
-ENV{ID_FS_UUID_ENC}=="?*", SYMLINK+="disk/by-uuid/$env{ID_FS_UUID_ENC}"
-SUBSYSTEM=="block", ACTION=="add|change", ENV{ID_FS_TYPE}=="bcache", \
+ENV{BCACHE_ID_FS_UUID_ENC}=="?*", SYMLINK+="disk/by-uuid/$env{BCACHE_ID_FS_UUID_ENC}"
+SUBSYSTEM=="block", ACTION=="add|change", ENV{BCACHE_ID_FS_TYPE}=="bcache", \
         RUN+="bcache-register $tempnode"
 LABEL="bcache_backing_end"
 
--- bcache-tools-20130909/probe-bcache.c.udevfix-old	2013-09-30 19:23:00.663897443 +0200
+++ bcache-tools-20130909/probe-bcache.c	2013-09-30 19:24:13.260847312 +0200
@@ -61,9 +61,9 @@
 		uuid_unparse(sb.uuid, uuid);
 
 		if (udev)
-			printf("ID_FS_UUID=%s\n"
-			       "ID_FS_UUID_ENC=%s\n"
-			       "ID_FS_TYPE=bcache\n",
+			printf("BCACHE_ID_FS_UUID=%s\n"
+			       "BCACHE_ID_FS_UUID_ENC=%s\n"
+			       "BCACHE_ID_FS_TYPE=bcache\n",
 			       uuid, uuid);
 		else
 			printf("%s: UUID=\"\" TYPE=\"bcache\"\n", uuid);
