Index: debian/cassandra.install
===================================================================
--- debian/cassandra.install	(revision 915954)
+++ debian/cassandra.install	(working copy)
@@ -6,14 +6,14 @@
 bin/nodeprobe usr/bin
 lib/antlr-3.1.3.jar usr/share/cassandra
 lib/clhm-production.jar usr/share/cassandra
-lib/commons-cli-1.1.jar usr/share/cassandra
-lib/commons-collections-3.2.1.jar usr/share/cassandra
-lib/commons-lang-2.4.jar usr/share/cassandra
-lib/google-collect-1.0-rc1.jar usr/share/cassandra
+build/lib/jars/commons-cli-1.1.jar usr/share/cassandra
+build/lib/jars/commons-collections-3.2.1.jar usr/share/cassandra
+build/lib/jars/commons-lang-2.4.jar usr/share/cassandra
+build/lib/jars/google-collections-1.0.jar usr/share/cassandra
 lib/high-scale-lib.jar usr/share/cassandra
-lib/jline-0.9.94.jar usr/share/cassandra
-lib/json_simple-1.1.jar usr/share/cassandra
+build/lib/jars/jline-0.9.94.jar usr/share/cassandra
+build/lib/jars/json-simple-1.1.jar usr/share/cassandra
 lib/libthrift-r894924.jar usr/share/cassandra
-lib/log4j-1.2.15.jar usr/share/cassandra
-lib/slf4j-api-1.5.8.jar usr/share/cassandra
-lib/slf4j-log4j12-1.5.8.jar usr/share/cassandra
+build/lib/jars/log4j-1.2.14.jar usr/share/cassandra
+build/lib/jars/slf4j-api-1.5.8.jar usr/share/cassandra
+build/lib/jars/slf4j-log4j12-1.5.8.jar usr/share/cassandra
