--- Makefile.orig	2011-02-16 09:43:20.512669276 +0500
+++ Makefile	2011-02-16 09:43:50.669808342 +0500
@@ -65,8 +65,8 @@
 	cp -r ebin src include lib Emakefile $(DESTROOT)$(ERLDIR)/
 	install -p -D -m 0755 contrib/reverse_mpegts $(DESTROOT)/usr/bin/reverse_mpegts
 	install -p -D -m 0755 contrib/erlyctl.debian $(DESTROOT)/usr/bin/erlyctl
-	mkdir -p $(DESTROOT)/etc/init.d/
-	ln -s /usr/bin/erlyctl $(DESTROOT)/etc/init.d/erlyvideo
+	mkdir -p $(DESTROOT)/etc/rc.d/init.d/
+	ln -s /usr/bin/erlyctl $(DESTROOT)/etc/rc.d/init.d/erlyvideo
 	cp -r wwwroot $(DESTROOT)/var/lib/erlyvideo/
 	mkdir -p $(DESTROOT)/var/log/erlyvideo
 	install -p -D -m 0644 priv/erlyvideo.conf.debian $(DESTROOT)/etc/erlyvideo/erlyvideo.conf
