--- Makefile.orig	2011-02-15 09:54:06.000000000 +0500
+++ Makefile	2011-02-16 09:36:36.595540028 +0500
@@ -63,17 +63,15 @@
 	mkdir -p $(DESTROOT)/var/lib/erlyvideo/plugins
 	mkdir -p $(DESTROOT)$(ERLDIR)
 	cp -r ebin src include lib Emakefile $(DESTROOT)$(ERLDIR)/
-	mkdir -p $(DESTROOT)/usr/bin/
-	cp contrib/reverse_mpegts $(DESTROOT)/usr/bin/reverse_mpegts
-	cp contrib/erlyctl.debian $(DESTROOT)/usr/bin/erlyctl
+	install -p -D -m 0755 contrib/reverse_mpegts $(DESTROOT)/usr/bin/reverse_mpegts
+	install -p -D -m 0755 contrib/erlyctl.debian $(DESTROOT)/usr/bin/erlyctl
 	mkdir -p $(DESTROOT)/etc/init.d/
 	ln -s /usr/bin/erlyctl $(DESTROOT)/etc/init.d/erlyvideo
 	cp -r wwwroot $(DESTROOT)/var/lib/erlyvideo/
 	mkdir -p $(DESTROOT)/var/log/erlyvideo
-	mkdir -p $(DESTROOT)/etc/erlyvideo
-	cp priv/erlyvideo.conf.debian $(DESTROOT)/etc/erlyvideo/erlyvideo.conf
-	cp priv/log4erl.conf.debian $(DESTROOT)/etc/erlyvideo/log4erl.conf
-	cp priv/production.config.debian $(DESTROOT)/etc/erlyvideo/production.config
+	install -p -D -m 0644 priv/erlyvideo.conf.debian $(DESTROOT)/etc/erlyvideo/erlyvideo.conf
+	install -p -D -m 0644 priv/log4erl.conf.debian $(DESTROOT)/etc/erlyvideo/log4erl.conf
+	install -p -D -m 0644 priv/production.config.debian $(DESTROOT)/etc/erlyvideo/production.config
 	mkdir -p $(DESTROOT)/var/cache/erlyvideo/licensed
 	for i in deps/amf deps/log4erl deps/erlydtl deps/erlmedia deps/mpegts deps/rtmp deps/rtp deps/rtsp deps/ibrowse ; do (cd $$i; make DESTROOT=$(DESTROOT) ERLANG_ROOT=$(ERLANG_ROOT) VERSION=$(VERSION) install) ; done
 
