From 59ccc22e5f437d64961bc7f17763f559a038152e Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Sun, 31 Oct 2010 11:52:02 +0300
Subject: [PATCH 4/4] Fix installing of contributed rtmp apps

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 deps/rtmp/Makefile |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/deps/rtmp/Makefile b/deps/rtmp/Makefile
index 22f5627..7f9ab80 100644
--- a/deps/rtmp/Makefile
+++ b/deps/rtmp/Makefile
@@ -30,9 +30,8 @@ clean-doc:
 install:
 	mkdir -p $(DESTROOT)$(ERLDIR)/ebin
 	mkdir -p $(DESTROOT)/usr/bin
-	mkdir -p $(DESTROOT)$(ERLDIR)/contrib
 	mkdir -p $(DESTROOT)$(ERLDIR)/include
-	install -c -m 755 contrib/* $(DESTROOT)$(ERLDIR)/contrib
+	install -c -m 755 contrib/so_bench $(DESTROOT)/usr/bin/so_bench
 	install -c -m 755 contrib/rtmp_bench $(DESTROOT)/usr/bin/rtmp_bench
 	install -c -m 644 ebin/*.beam $(DESTROOT)$(ERLDIR)/ebin
 	install -c -m 644 ebin/*.app $(DESTROOT)$(ERLDIR)/ebin
-- 
1.7.2.3

